packages/block-editor/tsconfig.json

```diff
@@ -0,0 +1,13 @@
+{
+	"extends": "../../tsconfig.base.json",
+	"compilerOptions": {
+		"rootDir": "src",
+		"declarationDir": "build-types"
+	},
+	// NOTE: This package is being progressively typed. You are strongly
+	// encouraged to expand this array with files which can be type-checked. At
+	// some point in the future, this can be simplified to `src/**/*`.
+	"include": [
+		"src/utils/dom.js"
+	]
```

Comment:
If this were a fully typed package, it should have all of the `@wordpress` dependencies listed here as `references`. Essentially a reproduction of this list:

https://github.com/WordPress/gutenberg/blob/c72030189017c8aac44453c1386f4251e45e80df/packages/block-editor/package.json#L26-L46

🤔 Not all of those are typed and I'm not sure what the best approach is to make sure this doesn't sneak up on us in the future. Using project references correctly ensures that the TypeScript compiler (`tsc`) builds projects in the correct order and prohibits circular dependencies. If we omit some references that are required, we might bump into cases where we se intermittent failures in some builds because `tsc` doesn't build projects in the correct order.

I think at least we should add the typed packages and a comment here.

Reply:
> If this were a fully typed package, it should have all of the `@wordpress` dependencies listed here as `references`.
>...
>Not all of those are typed
>...
>I think at least we should add the typed packages and a comment here.

I could test to see if it successfully builds with all of them included as references. If it would only become a "problem" at the point where we try to opt-in a new file that _uses_ those dependencies, I think that's going to be a problem in any case.

But it depends if it can successfully build at all by the mere _presence_ of those references.

If it can't, then I'll just add the typed packages.

(Unless you think we should do that anyways)

Reply:
d4640a5f98 It seems to tolerate it.

Aside: It may be fair to assume we should want to _enforce_ that `@wordpress/` dependencies of a package's `package.json` should be listed in `tsconfig.json` as `references`? Seems automatable.

Reply:
Oh, duh, I did this before addressing https://github.com/WordPress/gutenberg/pull/21362#issuecomment-608109744 . It's not quite as fond after those changes.

Reply:
Reverted in 7cac6105df.